packages/components/src/ConfirmationModal/ConfirmationModal.tsx

```diff
@@ -107,6 +107,14 @@ interface BaseConfirmationModalProps {
    */
   readonly cancelLabel?: string;
 
+  /**
+   * Type (Work or destructive) for confirm button.
+   */
+  readonly type?: Exclude<
+    BaseActionProps["variation"] | DestructiveActionProps["variation"],
+    "learning"
+  >;
```

Comment:
the more i think about this, I don't think we actually want to couple this to the button types. The reason being is the same as my comment around the naming `actionType`. We may in the future want the `type` to represent something other than just the button variation. I think we should actaully just go with `"work" | "destructive"` here.

I was also thinking that this may make more sense to be named `variation` instead of `type`.

Thoughts?

Reply:
Variation maps more closely to the way we describe the work and destructive options of Button, seems reasonable

Reply:
I would like to keep it simple too ("work" | "destructive") but this variation value is assigned to the primary action of modal (which is button variation), if we could find a way around it that would be awesome.